michaeltorres1/Expense-Tracker
# Create expense store

Added the expense store and is not working with expenses dynamically adding to a wallet.

### Changes made
1. No longer using the `DummyExpenses` component
2. Added a `WalletContext` and `WalletProvider` to house the expenses
3. Expense are dynamic but exists only for the current session being that it is only the client we are working on for now

This PR only resolves the issue of having expenses in a wallet and it being dynamic via providing our components the context for our wallet. A lot more logic needs to be implemented in a separate PR and one thing I did not cover in this PR is expenses aren't being filtered by different wallets and currently all expenses go only in one wallet. The concept of the wallet is just one wallet for now and in a separate PR we need to handle this too. This is just a heads up when testing this PR.

### To test this PR follow the following steps
1. go to `localhost:3000/register`
2. register a user (email must contain `@` and password must be more than 6 characters for both password and confirm password inputs)
3. passwords do not have to match for now
4. should take you to `localhost:3000/wallet` immediately after registering
5. add an expense and the expense should be seen immediately on top of the form
